project-2-e-commerce/src/admin/Orders.js

```diff
@@ -0,0 +1,135 @@
+import { adminSelector, getOrders, updateOrdersAdmin } from '../store/slices/AdminSlice'
+import { useDispatch, useSelector } from 'react-redux'
+import { useState, useEffect } from 'react'
+import { getSortFilter, getOrdersOption } from './helpers'
+import { useTranslation } from 'react-i18next'
+import Select from 'react-select'
+import LoadingComponent from '../component/loading'
+import { creatAddressText, formatPrice, getOrderStatusStyle } from '../helpers'
+
+function Orders() {
+    const dispatch = useDispatch()
+    const { ordersList, filterOrders, isLoading } = useSelector(adminSelector)
+    const [inputSearch, setInputSearch] = useState('')
+    const { t } = useTranslation()
+
+    useEffect(()=>{
+        dispatch(getOrders())
+    }, [])
+
+    const handleSearch = (e) => {
+        setInputSearch(e.target.value)
+        const filter = {...filterOrders, search: e.target.value}
+        dispatch(getOrders(filter))
+    }
+
+    const handleChange = (e) => {
+        const { sort, order } = getSortFilter(e.value)
+        const filter = {...filterOrders, sort, order}
+        dispatch(getOrders(filter))
+    }
+
+    const handleApprove = async (order) => {
+        const newOderItem = {...order, status: 'approved'}
+        await dispatch(updateOrdersAdmin({id: order.id, newOderItem})).unwrap()
+        dispatch(getOrders(filterOrders))
+    }
+
+    const handleRefuse = async (order) => {
+        const newOderItem = {...order, status: 'refuse'}
+        await dispatch(updateOrdersAdmin({id: order.id, newOderItem})).unwrap()
+        dispatch(getOrders(filterOrders))
+    }
```

Comment:
2 hàm này là 1 nè anh, truyền thêm 1 param là status thôi nữa là được
`handleStatus = async (order, status)`
Nhưng update thì dùng PATCH chỉ update mỗi cái status dựa trên ID thôi là được rồi mà nhỉ, đâu cần phải lấy toàn bộ thằng order đâu anh